### ISSUE TITLE
AOT compilation of jitclass

### ISSUE THREAD

**Issue by zaheeroz** [numba/numba#6648]
I couldnt find a way to AOT compile a whole class. 
So I guess it will be great to have that feature. I know methods can be AOT compiled which is a great feature but classes will add so much to it.  
The use cases are many, one can compile a whole class of methods instead of one by one. Also, a great way to obfuscate the code.

**Comment by stuartarchibald** [comment]
Thanks for the report. Just to clarify what you are asking for, is it:

1. The ability to "automatically" AOT compile all the methods in a python class.
2. The ability to AOT compile a `numba.experimental.jitclass`
3. Something else?

Thanks.

**Comment by zaheeroz** [comment]
Thanks for replying so quickly. Yes I would like to be able to automatically compile (AOT ) a jitclass using a decorator like @cc.export('myClassName', 'spec'). A feature similar for methods used here https://numba.pydata.org/numba-doc/dev/user/pycc.html.  But instead which compiles (AOT) a jitclass.  When I say jitclass I am referring to a complete class that includes all its members/methods and variables etc.  So that one is able to import the class (object) and be able to use all its methods without the need to import individual methods.
Thanks.

**Comment by stuartarchibald** [comment]
@zaheeroz thanks for clarifying. Numba has a "jitclass" https://numba.readthedocs.io/en/stable/user/jitclass.html, whilst you cannot AOT compile it at present (it's not been implemented), I think that `@jitclass` is the "compilable" class concept you are looking for?

**Comment by zaheeroz** [comment]
yes,  I am looking for a compilable class. I already converted my python classes to jitclasses and I am happy with the performance but now need to compile them into binary form, like one would do in c/c++.

**Comment by stuartarchibald** [comment]
> yes, I am looking for a compilable class. I already converted my python classes to jitclasses and I am happy with the performance but now need to compile them into binary form, like one would do in c/c++.

Great, thanks for confirming, as mentioned above, this has not been implemented yet. I'll mark this as a feature request.

**Comment by guzuomuse** [comment]
waiting....

**Comment by gmarkall** [comment]
@guzuomuse If you need to get updates about a particular issue in your notifications, you can hit the "Subscribe" button in the right-hand column, which saves having to post a comment on the issue.
